vector/src/main/java/im/vector/app/features/voice/AbstractVoiceRecorder.kt

```diff
@@ -57,9 +61,22 @@ abstract class AbstractVoiceRecorder(
         }
     }
 
-    override fun startRecord() {
+    override fun initializeRecord(roomId: String, attachmentData: ContentAttachmentData) {
+        getFilenameFromUri(context, attachmentData.queryUri)?.let {
+            val voiceMessageFolder = File(outputDirectory, roomId.md5())
+            outputFile = File(voiceMessageFolder, it)
+        }
+    }
+
+    override fun startRecord(roomId: String) {
         init()
-        outputFile = File(outputDirectory, "Voice message.$filenameExt")
+        val fileName = "Voice message-${UUID.randomUUID()}.$filenameExt"
+        val outputDirectoryForRoom = File(outputDirectory, roomId.md5()).apply {
+            if (!exists()) {
+                mkdirs()
```

Comment:
`mkdirs()` do the existing check, do you remember :)

Reply:
https://github.com/vector-im/element-android/pull/4527/commits/cd4e9916ff076c8dcef60d3b619f362e28c7abff